### PR TITLE
ffmpeg deprecated functions

### PR DESCRIPTION
**Pull request by ferdymercury**
Attempt to fix https://github.com/ferdymercury/amide/issues/22

### CODE REVIEW

**Comment by emollier** [comment]
Good morning,

7180fc9e34a78f7ea3373bac1abf4b87388fcd23 looks good.

In 23a2596a5761dc2aaf042ebcbd1c791d95571867, what is supposed to happen when _pkt_ is _NULL_?  From what I interpret for the moment, if _av_packet_alloc_ returns a _NULL_ pointer, then the data writing is skipped and goes right to _av_packet_free_ and the function _mpeg_encode_frame_ returns _TRUE_.  I'm not sure this is what should happen.  Besides, it tastes like a double free condition may occur.

On the other hand, test with the prospective Debian package upgrade show the build is fixed with _ffmpeg_ 5.0.1.  I also managed to produce a fly through video with the example file _m2862-small.xif_, which gives an _mpg_ file with the expected video inside, so this is very encouraging and helpful.  Thank you!

Have a nice day!  :)

File: amide-current/src/mpeg_encode.c [hunk 1]
```diff
@@ -372,23 +374,41 @@ gpointer mpeg_encode_setup(gchar * output_filename, mpeg_encode_t type, gint xsi
...
+  int result = avcodec_send_frame(encode->context, encode->picture);
+  if (result == AVERROR_EOF)
+    return TRUE;
+  else if (result < 0)
+    return FALSE;
+  else { 
+    AVPacket* pkt = av_packet_alloc();
+    if (pkt != NULL) {
+      while (avcodec_receive_packet(encode->context, pkt) == 0) {
+	fwrite(pkt->data, 1, pkt->size, encode->output_file);
```
Comment (emollier) [inline amide-current/src/mpeg_encode.c:404]: Minor nipicks: most of the code looks space indented, but this line seems to have a tab, also some spaces may be trimmed at the end of a couple of lines.

Comment (ferdymercury) [reply amide-current/src/mpeg_encode.c]: good catch

**Comment by ferdymercury** [comment]
> Good morning,
> 
> [7180fc9](https://github.com/ferdymercury/amide/commit/7180fc9e34a78f7ea3373bac1abf4b87388fcd23) looks good.
> 
> In [23a2596](https://github.com/ferdymercury/amide/commit/23a2596a5761dc2aaf042ebcbd1c791d95571867), what is supposed to happen when _pkt_ is _NULL_? From what I interpret for the moment, if _av_packet_alloc_ returns a _NULL_ pointer, then the data writing is skipped and goes right to _av_packet_free_ and the function _mpeg_encode_frame_ returns _TRUE_. I'm not sure this is what should happen. Besides, it tastes like a double free condition may occur.

good catch again!

**Comment by emollier** [comment]
It's all good on my end.  I pulled your changes as _quilt_ patch for the Debian package until the next release of amide.

Many thanks!  :)

**Comment by enlf-dev** [comment]
All is fine on MSYS2.

**Review by enlf-dev — COMMENTED** [review]
Seems to work on MSYS2 as well; still with ffmpeg 4.

**Pull request merged**